gen/enum.go

```diff
@@ -101,6 +106,96 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 			<end>
 			return fmt.Sprintf("<$enumName>(%d)", <$w>)
 		}
+
+		func (<$v> <$enumName>) MarshalJSON() ([]byte, error) {
```

Comment:
Should $v be parameterized? The variable name is scoped to the function. Does giving this variable freedom introduce the possibility of a name collision?